src/main/java/net/barrage/school/java/ecatalog/app/ProductServiceImpl.java

```diff
@@ -33,6 +33,18 @@ public List<Product> listProducts() {
                 .toList();
     }
 
+    @SneakyThrows
+    @Override
+    public List<Product> searchProducts(String query) {
+        return objectMapper.readValue(productsSourceFile, SourceProductList.class).stream()
```

Comment:
minor: Why not to re-use `listProducts`? It turns that u copy-pasted some logic u don't really need.